provider/datagen/src/transform/cldr/currency/mod.rs

```diff
@@ -53,29 +52,43 @@ fn which_currency_pattern(
         Err(icu_properties::PropertiesError::PropDataLoad(e)) => {
             return Err(e);
         }
-        Err(_) => unreachable!(),
+        Err(_) => unreachable!("load_for_general_category_group should only return PropDataLoad"),
     };
 
-    let right_letter_test_position = {
-        if currency_sign_ind < first_num_ind && currency_sign_ind < last_num_ind {
-            true
-        } else if currency_sign_ind > first_num_ind && currency_sign_ind > last_num_ind {
-            false
+    #[derive(PartialEq)]
+    enum PlaceHolderCharCloserToNumber {
+        First,
+        Last,
+    }
+
+    let char_closer_to_number = {
+        if currency_sign_index < first_num_index {
+            PlaceHolderCharCloserToNumber::Last
+        } else if currency_sign_index > last_num_index {
+            PlaceHolderCharCloserToNumber::First
         } else {
-            panic!("The currency sign is in the middle of the pattern.")
+            return Err(DataError::custom(
+                "Currency sign is in the middle of the pattern",
+            ));
         }
     };
 
-    if right_letter_test_position {
-        // Check if the last character has the property of letter (L).
-        return Ok(!letters_set
+    if char_closer_to_number == PlaceHolderCharCloserToNumber::Last {
```

Comment:
Suggestion: it's more conventional in Rust to `match` on enum variants. If you do this, you don't need `PartialEq`, and it lets the compiler optimize more.

Reply:
Thanks for the info, it is now simplified.